pkg/controller/certificates/rootcacertpublisher/publisher.go

```diff
@@ -58,18 +58,18 @@ func NewPublisher(cmInformer coreinformers.ConfigMapInformer, nsInformer coreinf
 		queue:  workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "root_ca_cert_publisher"),
 	}
 
-	cmInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+	cmHandler, _ := cmInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		DeleteFunc: e.configMapDeleted,
 		UpdateFunc: e.configMapUpdated,
 	})
 	e.cmLister = cmInformer.Lister()
-	e.cmListerSynced = cmInformer.Informer().HasSynced
+	e.cmListerSynced = cmHandler.HasSynced
```

Comment:
This has changed from "is my lister synced" to "is my lister synced and my event handler been synced"? Why change the precondition for starting on this controller to include event handler syncing?

Reply:
I may have gotten a little overeager on a few of the controllers... I already backed out some. It turns out we just don't actually wait for them to be synced as much as I thought we did